s3fs/core.py

```diff
@@ -1209,6 +1214,16 @@ def __init__(self, s3, path, mode='rb', block_size=5 * 2 ** 20, acl="",
                     self.version_id = info.get('VersionId')
             except (ClientError, ParamValidationError) as e:
                 raise_from(IOError("File not accessible", path), e)
+                if self.file_backed_cache:
+                    fd = tempfile.TemporaryFile()
+                    fd.seek(self.size - 1)
+                    fd.write(b'1')
+                    fd.flush()
+                    f_no = fd.fileno()
+                    self.start = 0
+                    self.end = self.size
+                    self.cache = mmap.mmap(f_no, self.size)
```

Comment:
Note that the windows version of mmap will automatically extend the file size here, rather than doing the sparse trick (but this might take up the fill space of the file - to be tried). I suspect the sparse trick might not work on windows, you'd get "write past EOF" error or similar.

Reply:
What is windows?  lol. sorry i only have mac and linux at hand.